contribs/drt/src/main/java/org/matsim/contrib/drt/optimizer/rebalancing/targetcalculator/DemandEstimatorWithCustomMappingFunction.java

```diff
@@ -0,0 +1,45 @@
+package org.matsim.contrib.drt.optimizer.rebalancing.targetcalculator;
+
+import org.matsim.contrib.drt.analysis.zonal.DrtZone;
+import org.matsim.contrib.drt.optimizer.rebalancing.demandestimator.ZonalDemandEstimator;
+import org.matsim.contrib.dvrp.fleet.DvrpVehicle;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.ToDoubleFunction;
+
+public class DemandEstimatorWithCustomMappingFunction implements RebalancingTargetCalculator {
+	private final ZonalDemandEstimator demandEstimator;
+	private final double demandEstimationPeriod;
+
+	public DemandEstimatorWithCustomMappingFunction(ZonalDemandEstimator demandEstimator, double demandEstimationPeriod) {
+		this.demandEstimator = demandEstimator;
+		this.demandEstimationPeriod = demandEstimationPeriod;
+	}
+
+	@Override
+	public ToDoubleFunction<DrtZone> calculate(double time, Map<DrtZone, List<DvrpVehicle>> rebalancableVehiclesPerZone) {
+		return transform(time);
+	}
+
+	/**
+	 * Custom transform function (simple example)
+	 */
+	private ToDoubleFunction<DrtZone> transform(double time) {
+		ToDoubleFunction<DrtZone> originalFunction = demandEstimator.getExpectedDemand(time, demandEstimationPeriod);
+
+		double lookAheadTime = 300; //TODO make this a variable (or read from config file: = rebalancing period)
+		ToDoubleFunction<DrtZone> originalFunctionWithLookAhead = demandEstimator.getExpectedDemand(time + lookAheadTime, demandEstimationPeriod);
```

Comment:
The current implementation of `ZonalDemandEstimator` discretizes predicted demand time-wise and always takes a value from one time bin (no interpolation is done). Therefore, if `lookAheadTime` is similar to or smaller than the time bin, we may not observe a significant changes in the behaviour.

What if we:
- add interpolation to the zonal demand estimation?
- (or alternatively) change the "max of two" function into the "average of two"? (line 35)

Reply:
The look ahead time can be helpful for the requests submitted at the start of the time bin (e.g. time bin size = 1800, then the trips submitted right after xx:00 and xx:30 may need to wait longer as there is not yet any rebalnacing vehicles approaching). If we have, say 5 minutes, looking ahead time, then those requests can wait shorter. This can be quite helpful especially for the rural area scenario, where the vehicles density are relatively thin. 

Yeah, I think interpolation is a good idea.

Alternatively, we can also make the demand estimator only return 0 or 1. This is also the original idea from Kai: keeping 1 vehilce in each village (with potential request). And as long as the rebalancing period is short engouh and the zonal system is not too large, then it should provide a good performance. And this can also make the "max" operation more suitable.